src/pantab/writer.cpp

```diff
@@ -519,7 +519,8 @@ void write_to_hyper(
     const std::map<SchemaAndTableName, nb::capsule> &dict_of_capsules,
     const std::string &path, const std::string &table_mode,
     nb::iterable not_null_columns, nb::iterable json_columns,
-    nb::iterable geo_columns) {
+    nb::iterable geo_columns,
+    const int8_t version) {
```

Comment:
Since the HyperProcess constructor expects all parameters to be strings, I think you can just declare a parameter here of `const std::map<std::string, std::string> process_params` and pass that to the HyperProcess constructor. 

To make it easier you can map all of the parameter values to strings in Python before sending through to the extension